### PR TITLE
Add single node benchmark

### PR DESCRIPTION
**Pull request by lxfind**
## Description 

See details in README.

## Test Plan 

CI

---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [ ] protocol change
- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

File: crates/sui-single-node-benchmark/src/execution.rs [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+        .generate_transactions(Arc::new(NonMoveTxGenerator::new()))
+        .await;
+    benchmark_transactions(&ctx, transactions).await;
+}
+
+/// Benchmark Move transactions.
+/// Each transaction is a programmable transaction that performs a series of operations specified
+/// by the parameters.
+///
+/// \tx_count: the number of transactions to execute.
```
Comment (mwtian) [inline crates/sui-single-node-benchmark/src/execution.rs:30]: Maybe include the `end_to_end` arg too for consistency.

File: crates/sui-single-node-benchmark/src/main.rs [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+use clap::Parser;
+use sui_single_node_benchmark::command::Command;
+use sui_single_node_benchmark::execution::{
+    benchmark_move_transactions, benchmark_simple_transfer,
+};
+
+#[tokio::main]
+async fn main() {
+    let _guard = telemetry_subscribers::TelemetryConfig::new()
+        .with_log_level("off,sui_single_node_benchmark=info")
```
Comment (mwtian) [inline crates/sui-single-node-benchmark/src/main.rs:13]: Does this override `RUST_LOG` env var?

**Review by mwtian — APPROVED** [review]

File: crates/sui-single-node-benchmark/move_package/sources/benchmark.move [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+            let coin = vector::pop_back(&mut coins);
+            let balance = coin::into_balance(coin);
+            balance::join(result_balance, balance);
+        };
+        vector::destroy_empty(coins);
+        result
+    }
+
+    public fun run_computation(num: u64) {
+        let results = vector<u64>[];
```
[the remark this answers is not in the document]
Comment (lxfind) [reply crates/sui-single-node-benchmark/move_package/sources/benchmark.move:32]: Yes, to exercise memory usage

File: crates/sui-single-node-benchmark/src/main.rs (hunk 2 above)
Comment (lxfind) [reply crates/sui-single-node-benchmark/src/main.rs:13]: No, the next line (with_env) will override log level if it contains RUST_LOG

**Pull request merged**